amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+        high_x = src_addr.dim_x - shift_x_u1;   // U1 addressing rounds down imagepatch start_x to nearest byte boundary
+        high_y = src_addr.dim_y;
+
+        if (border->mode == VX_BORDER_UNDEFINED)
+        {
+            low_x  += (vx_uint32)rx0;
+            low_y  += (vx_uint32)ry0;
+            high_x -= (vx_uint32)rx1;
+            high_y -= (vx_uint32)ry1;
+            vxAlterRectangle(&rect, (vx_int32)rx0, (vx_int32)ry0, -(vx_int32)rx1, -(vx_int32)ry1);
+        }
+
+        for (y = low_y; y < high_y; y++)
+        {
+            for (x = low_x; x < high_x; x++)
+            {
+                vx_uint32 xShftd = x + shift_x_u1;      // Bit-shift for U1 valid region start
+                vx_uint8 *dst_ptr = (vx_uint8*)vxFormatImagePatchAddress2d(dst_base, xShftd, y, &dst_addr);
+                vx_int32 count = (vx_int32)readMaskedRectangle(src_base, &src_addr, border, format, xShftd, y, (vx_uint32)rx0, (vx_uint32)ry0, (vx_uint32)rx1, (vx_uint32)ry1, m, v, shift_x_u1);
+
+                qsort(v, count, sizeof(vx_uint8), vx_uint8_compare);
+
+                switch (func)
+                {
+                case VX_NONLINEAR_FILTER_MIN:    res_val = v[0];         break; /* minimal value */
+                case VX_NONLINEAR_FILTER_MAX:    res_val = v[count - 1]; break; /* maximum value */
+                case VX_NONLINEAR_FILTER_MEDIAN: res_val = v[count / 2]; break; /* pick the middle value */
+                }
+                if (format == VX_DF_IMAGE_U1)
+                {
+                    *dst_ptr = (*dst_ptr & ~(1 << (xShftd % 8))) | (res_val << (xShftd % 8));
+                }
+                else
+                    *dst_ptr = res_val;
+            }
+        }
+    }
+
+    status |= vxCommitImagePatch(src, NULL, 0, &src_addr, src_base);
+    status |= vxCommitImagePatch(dst, &rect, 0, &dst_addr, dst_base);
+    
+    return status;
+}
+
+#define C_MAX_CONVOLUTION_DIM (15)
+
+static const vx_int16 gaussian5x5[5][5] =
+{
+    {1,  4,  6,  4, 1},
+    {4, 16, 24, 16, 4},
+    {6, 24, 36, 24, 6},
+    {4, 16, 24, 16, 4},
+    {1,  4,  6,  4, 1}
+};
+
+static const vx_uint32 gaussian5x5scale = 256;
+
+static vx_convolution vxCreateGaussian5x5Convolution(vx_context context)
+{
+    vx_convolution conv = vxCreateConvolution(context, 5, 5);
+    vx_status status = vxCopyConvolutionCoefficients(conv, (vx_int16 *)gaussian5x5, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    if (status != VX_SUCCESS)
+    {
+        vxReleaseConvolution(&conv);
+        return NULL;
+    }
+
+    status = vxSetConvolutionAttribute(conv, VX_CONVOLUTION_SCALE, (void *)&gaussian5x5scale, sizeof(vx_uint32));
+    if (status != VX_SUCCESS)
+    {
+        vxReleaseConvolution(&conv);
+        return NULL;
+    }
+    return conv;
+}
+
+static vx_status ownCopyImage(vx_image input, vx_image output)
+{
+    vx_status status = VX_SUCCESS; // assume success until an error occurs.
+    vx_uint32 p = 0;
+    vx_uint32 y = 0, x = 0;
```

Comment:
Codacy found an issue: [Variable 'y' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5686893)